File: 卡码/代码随想录/JS/动态规划/05-子序列问题/02-子序列(连续)/LeetCode_0053.js
//53. 最大子数组和
const maxSubArray = function(nums) {
  const dp=new Array(nums.length).fill(0);
	let res=nums[0];
	dp[0]=nums[0];
	for(let i=1;i<nums.length;i++){
		dp[i]=Math.max(dp[i-1]+nums[i],nums[i]);
		res=dp[i]>res?dp[i]:res;
	}
	return res;
};

File: 卡码/代码随想录/Go/动态规划/05-子序列问题/02-子序列(连续)/LeetCode_0053.go
// 53. 最大子数组和
package main

import (
	"math"
)

func maxSubArray(nums []int) int {
	dp := make([]int, len(nums))
	dp[0] = nums[0]
	res := nums[0]
	for i := 1; i < len(nums); i++ {
		dp[i] = int(math.Max(float64(dp[i-1]+nums[i]), float64(nums[i])))
		if dp[i] > res {
			res = dp[i]
		}
	}
	return res
}
